kubernetes/internal/create-weblogic-domain-job-template.yaml

```diff
@@ -416,28 +186,59 @@ data:
     set('NodeManagerPasswordEncrypted', admin_password)
 
     # Create a cluster
+    # ======================
     cd('/')
-    create(cluster_name, 'Cluster')
+    cl=create(cluster_name, 'Cluster')
+
+    if cluster_type == "CONFIGURED":
 
-    # Create managed servers
-    for index in range(0, number_of_ms):
-      cd('/')
+      # Create managed servers
+      for index in range(0, number_of_ms):
+        cd('/')
 
-      msIndex = index+1
-      name = '%MANAGED_SERVER_NAME_BASE%%s' % msIndex
+        msIndex = index+1
+        name = '%MANAGED_SERVER_NAME_BASE%%s' % msIndex
 
-      create(name, 'Server')
-      cd('/Servers/%s/' % name )
-      print('managed server name is %s' % name);
-      set('ListenAddress', '%DOMAIN_UID%-%s' % name)
-      set('ListenPort', server_port)
-      set('NumOfRetriesBeforeMSIMode', 0)
-      set('RetryIntervalBeforeMSIMode', 1)
-      set('Cluster', cluster_name)
+        create(name, 'Server')
+        cd('/Servers/%s/' % name )
+        print('managed server name is %s' % name);
+        set('ListenAddress', '%DOMAIN_UID%-%s' % name)
+        set('ListenPort', server_port)
+        set('NumOfRetriesBeforeMSIMode', 0)
+        set('RetryIntervalBeforeMSIMode', 1)
+        set('Cluster', cluster_name)
 
-      create(name,'Log')
-      cd('/Servers/%s/Log/%s' % (name, name))
-      set('FileName', '/shared/logs/%s.log' % name)
+        create(name,'Log')
+        cd('/Servers/%s/Log/%s' % (name, name))
+        set('FileName', '/shared/logs/%s.log' % name)
+    else:
+      print('Configuring Dynamic Cluster %s' % cluster_name)
+
+      templateName = cluster_name + "-template"
+      print('Creating Server Template: %s' % templateName)
+      st1=create(templateName, 'ServerTemplate')
+      print('Done creating Server Template: %s' % templateName)
+      cd('/ServerTemplates/%s' % templateName)
+      cmo.setListenPort(server_port)
+      cmo.setListenAddress('%DOMAIN_UID%-%MANAGED_SERVER_NAME_BASE%${id}')
+      cmo.setCluster(cl)
+      print('Done setting attributes for Server Template: %s' % templateName);
+
+
+      cd('/Clusters/%s' % cluster_name)
+      create(cluster_name, 'DynamicServers')
+      cd('DynamicServers/%s' % cluster_name)
+      set('ServerTemplate', st1)
+      set('ServerNamePrefix', "%MANAGED_SERVER_NAME_BASE%")
+      set('DynamicClusterSize', number_of_ms)
+      set('MaxDynamicClusterSize', 800)
```

Comment:
Remove commented out set calls.  Should we remove setting MaxDynamicClusterSize?

Reply:
In effect, D-C-S sets the number of ms that show up in the bean tree (virtual config beans of some kind), which in effect designates the servers that can be booted.   For example, when DCS is 10, you will see 10 managed servers in the config tree on the console even when they aren't booted.    It also can do 'double duty' of letting the wlst-scaling/external frameworks know the number of servers you'd like to be running.    Max DCS, as far as I know, simply prevents DCS from being configured higher than the specified value - I personally don't think it has much utility.   JMS and JTA usually require that DCS never be lowered during a scale down (instead just shutdown the servers you don't want) -- this lets these services iterate over the aforementioned 'virtual' beans to locate data that may be trapped in servers that are no longer going to be booted.

Reply:
This is being replaced with set('MaxDynamicClusterSize', number_of_ms). Also will be adding some validation logic to compare replicas and max dynamic cluster size in WlsClusterConfig class and make sure the max dynamic cluster size is honored and we won't scale above it.